strings/base_lock.h

```diff
@@ -118,4 +118,29 @@ WINRT_EXPORT namespace winrt
     private:
         impl::condition_variable m_cv{};
     };
+
+    struct slim_event
+    {
+        slim_event(slim_event const&) = delete;
+        slim_event const& operator=(slim_event const&) = delete;
+        slim_event() noexcept = default;
+
+        void signal() noexcept
+        {
+            signaled = true;
+            WINRT_WakeByAddressAll(&signaled);
+        }
+
+        void wait() noexcept
+        {
+            while (!signaled)
+            {
+                bool not_ready = false;
+                WINRT_WaitOnAddress(&signaled, &not_ready, sizeof(not_ready), 0xFFFFFFFF); // INFINITE
+            }
+        }
+
+    private:
+        bool signaled = false;
```

Comment:
May need to use `std::atomic<>`(can get away with `memory_order_relaxed`)  in case architecture does not support single-byte updates.

Reply:
Wouldn't that introduce unnecessary memory barriers? Those tend to be more costly than a few extra bytes in memory. Perhaps uint32_t instead of bool is sufficient.

Reply:
Even with `uint32_t` we need a release barrier so that the compiler won't reorder the store to `m_queue` after the store to `m_ready.signaled`. If the reordering occurred, then the lambda could observe `signaled` and run ahead to complete the coroutine, and then the first thread crashes trying to write to an already-freed `m_queue`.

Reply:
Aha, I can use ` std::atomic_signal_fence(release)` which is purely a compiler reordering directive. No code is generated by a signal fence; it merely tells the compiler not to reorder across the fence.

Reply:
I think this means I should withdraw `slim_event` since it does not have the same memory consistency guarantees as a regular event. (Regular events are full barriers.)

Reply:
Sounds good.